File: src/Asm.MooBank.Web/MooBankApp/src/models/reports/TagTrendReport.ts
import { TagSettings } from "../Tag";
import { BaseReport } from "./BaseReport";
import { TrendPoint } from "./TrendPoint";

export interface TagTrendReport extends BaseReport {
    tagId: number,
    tagName: string,
    months: TrendPoint[],
    average: number,
    offsetAverage: number,
}

export interface TagTrendReportSettings extends Pick<TagSettings, "applySmoothing"> {

}

export const defaultSettings: TagTrendReportSettings = {
    applySmoothing: false,
}

File: src/Asm.MooBank.Modules.Reports/Models/TagTrendReport.cs
namespace Asm.MooBank.Modules.Reports.Models;

public record TagTrendReport : ReportBase
{
    public required int TagId { get; init; }

    public required string TagName { get; init; }

    public required IEnumerable<TrendPoint> Months { get; init; }
    public decimal Average { get; set; }
    public decimal? OffsetAverage { get; set; }
}
